addons/payment_stripe/controllers/main.py

```diff
@@ -55,6 +55,13 @@ def stripe_s2s_create_setup_intent(self, acquirer_id, **kwargs):
     def stripe_s2s_process_payment_intent(self, **post):
         return request.env['payment.transaction'].sudo().form_feedback(post, 'stripe')
 
+    @http.route('/payment/stripe/s2s/process_payment_error', type='json', auth='public', csrf=False)
+    def stripe_s2s_process_payment_error(self, **post):
+        transaction = request.env['payment.transaction'].search([('reference', '=', post['reference'])])
+        transaction.state = 'error'
```

Comment:
The route being public allows anyone to make an RPC call with an arbitrary transaction reference and set the related transaction in the 'error' state. It should probably take an `access_token` parameter or similar to allow only the user making the payment to update the state of the transaction.

Reply:
yep, but all the routes in the stripe controller have the same public configuration, isn't it also a security problem for these other routes to be public?